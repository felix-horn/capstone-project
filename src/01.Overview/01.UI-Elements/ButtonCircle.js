import PropTypes from 'prop-types'
import styled from 'styled-components/macro'

ButtonCircle.propTypes = {
  children: PropTypes.element.isRequired,
  onClick: PropTypes.func,
  variant: PropTypes.oneOf(['small', 'primary']),
  className: PropTypes.string,
}

export default function ButtonCircle({
  children,
  onClick,
  variant,
  className,
}) {
  return (
    <ButtonLayout
      onClick={onClick}
      variant={variant}
      className={className}
      data-testid="button"
    >
      {children}
    </ButtonLayout>
  )
}

const ButtonLayout = styled.div`
  box-shadow: var(--strong-box-shadow);
  border-radius: 100%;
  height: 50px;
  width: 50px;
  background-color: var(--white);
  display: grid;
  place-items: center;
  color: var(--CTA-blue);

  ${(props) =>
    props.variant === 'primary' &&
    `background-color: var(--CTA-blue); color: var(--white);`}

  ${(props) => props.variant === 'small' && `width: 40px; height: 40px;`}
`
