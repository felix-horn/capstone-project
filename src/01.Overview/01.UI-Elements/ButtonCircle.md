Standard

```js padded
import MenuIcon from '@material-ui/icons/Menu'
;<ButtonCircle onClick={() => {}} title="Button Lable">
  <MenuIcon />
</ButtonCircle>
```

Primary

```js padded
import ScannerIcon from '@material-ui/icons/CropFree'
;<ButtonCircle onClick={() => {}} variant="primary" title="Button Lable">
  <ScannerIcon />
</ButtonCircle>
```

Small

```js padded
import AddShopIcon from '@material-ui/icons/PostAdd'
;<ButtonCircle onClick={() => {}} variant="small" title="Button Lable">
  <AddShopIcon />
</ButtonCircle>
```
