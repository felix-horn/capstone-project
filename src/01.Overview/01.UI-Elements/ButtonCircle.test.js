import { render } from '@testing-library/react'
import user from '@testing-library/user-event'
import ButtonCircle from './ButtonCircle'
import AddShopIcon from '@material-ui/icons/PostAdd'

describe('ButtonCircle', () => {
  it('renders correctly', () => {
    const { container } = render(
      <ButtonCircle>
        <AddShopIcon />
      </ButtonCircle>
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('calls function onClick', () => {
    const onClickMock = jest.fn()
    const { getByTestId } = render(
      <ButtonCircle onClick={onClickMock}>
        <AddShopIcon />
      </ButtonCircle>
    )
    const button = getByTestId('button')
    user.click(button)
    expect(onClickMock).toHaveBeenCalledTimes(1)
  })

  it('displays the correct height for variant "small"', () => {
    const { getByTestId } = render(
      <ButtonCircle variant="small">
        <AddShopIcon />
      </ButtonCircle>
    )
    const button = getByTestId('button')
    const style = window.getComputedStyle(button)
    console.log(style)
    console.log(style.height)
    expect(style.height).toBe('40px')
  })

  xit('displays the correct color for variant "primary"', () => {
    const { getByTestId } = render(
      <ButtonCircle variant="primary">
        <AddShopIcon />
      </ButtonCircle>
    )
    const button = getByTestId('button')
    const style = window.getComputedStyle(button)
    console.log(style)
    //    backgroundColor is not available
    console.log(style.backgroundColor)
    expect(style.backgroundColor).toBe('var(--CTA-blu)')
    expect(button).toHaveStyle(`background: var(--CTA-blu)`)
  })
})
