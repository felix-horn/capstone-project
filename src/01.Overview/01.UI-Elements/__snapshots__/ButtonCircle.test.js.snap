// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonCircle renders correctly 1`] = `
.c0 {
  box-shadow: var(--strong-box-shadow);
  border-radius: 100%;
  height: 50px;
  width: 50px;
  background-color: var(--white);
  display: grid;
  place-items: center;
  color: var(--CTA-blue);
}

.c0.primary {
  background-color: var(--CTA-blue);
  color: var(--white);
}

.c0.small {
  height: 40px;
  width: 40px;
}

<div
  class="c0"
  data-testid="button"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M17 19.22H5V7h7V5H5c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2v-7h-2v7.22z"
    />
    <path
      d="M19 2h-2v3h-3c.01.01 0 2 0 2h3v2.99c.01.01 2 0 2 0V7h3V5h-3V2zM7 9h8v2H7zM7 12v2h8v-2h-3zM7 15h8v2H7z"
    />
  </svg>
</div>
`;
